File: Basic Question Bank/a010/a010.py
import math

def calculate_count_based_on_base(number, base):
    counter = 0
    while(number % base == 0):
        counter += 1
        number //= base
    return number, counter

number = eval(input())
info = []

# check 2 first
number, c = calculate_count_based_on_base(number, 2)
if c:
    info.append((2, c))
# then check the rest
upper_bound = int(math.sqrt(number)) + 1
for i in range(3, upper_bound, 2):
    if(i > number):
        break
    number, c = calculate_count_based_on_base(number, i)
    if c:
        info.append((i, c))
if number > 1:
    info.append((number, 1))

printed_first_element = False
for pair in info:
    if pair[1] > 1:
        if ~printed_first_element:
            print(str(pair[0]) + "^" + str(pair[1]), end="")
            printed_first_element = ~printed_first_element
        else:
            print(" * " + str(pair[0]) + "^" + str(pair[1]), end="")
    else:
        if ~printed_first_element:
            print(str(pair[0]), end="")
            printed_first_element = ~printed_first_element
        else:
            print(" * " + str(pair[0]), end="")

File: Basic Question Bank/a010/a010.java
import java.util.Scanner;

public class a010 {

    // returns the count ex: this_func(8, 2) -> 3
    static int calculate_count_by_base(int number, int base) {
        int counter = 0;
        while(number % base == 0) {
            counter++;
            number /= base;
        }
        return counter;
    }

    // append a elementat the end of an array
    static int[] append(int arr[], int target) {
        int[] returning_array = new int[arr.length + 1];
        for(int i = 0; i < arr.length; i++) {
            returning_array[i] = arr[i];
        }
        returning_array[arr.length] = target;
        return returning_array;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int number = sc.nextInt(), temp = number;
        int[] info = new int[0];

        // let the base 2 first
        if(calculate_count_by_base(number, 2) > 0) {
            info = append(info, 2);
            info = append(info, calculate_count_by_base(number, 2));
            number /= Math.pow(2, calculate_count_by_base(number, 2));
        }

        // then do the rest, +=2 makes it faster
        for(int i = 3; i < (int)Math.sqrt(temp) + 1; i+=2) {
            if(calculate_count_by_base(number, i) > 0) {
                info = append(info, i);
                info = append(info, calculate_count_by_base(number, i));
                number /= Math.pow(i, calculate_count_by_base(number, i));
            }
        }
        if(number > 1) {
            info = append(info, number);
            info = append(info, 1);
        }

        // print
        boolean first_printed = false;
        for(int i = 0; i < info.length; i+=2) {
            if(!first_printed) {
                if(info[i + 1] > 1) {
                    System.out.print(info[i] + "^" + info[i + 1]);
                    first_printed = true;
                }
                else {
                    System.out.print(info[i]);
                    first_printed = true;
                }
            }
            else {
                if(info[i + 1] > 1) {
                    System.out.print(" * " + info[i] + "^" + info[i + 1]);
                }
                else {
                    System.out.print(" * " + info[i]);
                }
            }
        }

        sc.close();
    }
}
